openssl-dynamic/src/main/c/ssl.c

```diff
@@ -334,6 +343,51 @@ static long bio_java_bytebuffer_ctrl(BIO* bio, int cmd, long num, void* ptr) {
     }
 }
 
+// This code is based on libcurl:
+// https://github.com/curl/curl/blob/curl-7_61_0/lib/vtls/openssl.c#L521
+#ifndef OPENSSL_NO_ENGINE
+/*
+ * Supply default password to the engine user interface conversation.
+ * The password is passed by OpenSSL engine from ENGINE_load_private_key()
+ * last argument to the ui and can be obtained by UI_get0_user_data(ui) here.
+ */
+static int ssl_ui_reader(UI *ui, UI_STRING *uis)
+{
+    const char *password;
+    switch (UI_get_string_type(uis)) {
+    case UIT_PROMPT:
+        // fall-through
```

Comment:
FWIW, I was fine without a comment here because the fallthrough was obvious and normal.

Reply:
I guess it cant hurt :)

Reply:
(I'm also fine without fall through here bcz it is expected stacking of cases with no-code ... having the comment brings attention to an area where attention isn't necessary)

Reply:
Ok will remove